subsys/nrf_security/src/drivers/cracen/cracenpsa/src/sign.c

```diff
@@ -193,14 +200,19 @@ static int cracen_signature_prepare_ec_pubkey(struct sitask *t, struct si_sig_pu
 	status = SX_ERR_INCOMPATIBLE_HW;
 
 	if (IS_ENABLED(PSA_NEED_CRACEN_PURE_EDDSA_TWISTED_EDWARDS)) {
+		if (PSA_KEY_LIFETIME_GET_LOCATION(psa_get_key_lifetime(attributes)) ==
+		    PSA_KEY_LOCATION_CRACEN_KMU) {
+			key_buffer = kmu_push_area;
+			key_buffer_size = sizeof(kmu_push_area);
+		}
 		if (alg == PSA_ALG_PURE_EDDSA) {
 			pubkey->def = si_sig_def_ed25519;
 
 			if (PSA_KEY_TYPE_IS_ECC_PUBLIC_KEY(psa_get_key_type(attributes))) {
 				pubkey->key.ed25519 = (struct sx_ed25519_pt *)key_buffer;
 				return SX_OK;
 			}
-			if (curvesz != key_buffer_size) {
+			if (curvesz >= key_buffer_size) {
 				return SX_ERR_INVALID_KEY_SZ;
 			}
 			pubkey->key.ed25519 = (struct sx_ed25519_pt *)pubkey_buffer;
```

Comment:
Missing:
calls to cracen_kmu_prepare_key and cracen_kmu_clean_key (or similar).
Note those are currently called inside a context where cryptomaster mutex is locked to ensure single usage of the buffer.

Maybe ed25519 keys should be pushed directly to correct location in pke-ram instead of the push area buffer.

Or maybe asymmetric should have a separate buffer.